server/src/main/java/org/elasticsearch/cluster/metadata/DataStreamGlobalRetention.java

```diff
@@ -8,31 +8,25 @@
 
 package org.elasticsearch.cluster.metadata;
 
-import org.elasticsearch.TransportVersion;
-import org.elasticsearch.TransportVersions;
-import org.elasticsearch.cluster.AbstractNamedDiffable;
-import org.elasticsearch.cluster.ClusterState;
-import org.elasticsearch.cluster.NamedDiff;
-import org.elasticsearch.common.collect.Iterators;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
 import org.elasticsearch.core.Nullable;
 import org.elasticsearch.core.TimeValue;
 import org.elasticsearch.features.NodeFeature;
 import org.elasticsearch.xcontent.ParseField;
-import org.elasticsearch.xcontent.ToXContent;
+import org.elasticsearch.xcontent.ToXContentFragment;
 import org.elasticsearch.xcontent.XContentBuilder;
 
 import java.io.IOException;
-import java.util.Iterator;
 import java.util.Objects;
 
 /**
  * A cluster state entry that contains global retention settings that are configurable by the user. These settings include:
  * - default retention, applied on any data stream managed by DSL that does not have an explicit retention defined
  * - max retention, applied on every data stream managed by DSL
  */
-public final class DataStreamGlobalRetention extends AbstractNamedDiffable<ClusterState.Custom> implements ClusterState.Custom {
+public final class DataStreamGlobalRetention implements Writeable, ToXContentFragment {
```

Comment:
I could convert this to a record, but I prefer to keep this change as minimal as possible. I can follow up with it at a follow up PR if we keep this class.

Reply:
Does this even need to implement ToXContentFragment any more? It's not written to the cluster state, and it's not directly written to a response anywhere, is it?

Reply:
Ah good point! I will remove it.